GoogleUtilities/Example/Tests/Swizzler/GULSceneDelegateSwizzlerTest.m

```diff
@@ -135,6 +147,149 @@ - (void)testNotificationCenterRegister {
   }
 }
 
+#pragma mark - Tests to test that Plist flag is honored
+
+/** Tests that app delegate proxy is enabled when there is no Info.plist dictionary. */
+- (void)testAppProxyPlistFlag_NoFlag {
+  // No keys anywhere. If there is no key, the default should be enabled.
+  NSDictionary *mainDictionary = nil;
+  id mainBundleMock = OCMPartialMock([NSBundle mainBundle]);
+  [[[mainBundleMock expect] andReturn:mainDictionary] infoDictionary];
+
+  XCTAssertTrue([GULSceneDelegateSwizzler isSceneDelegateProxyEnabled]);
+  [mainBundleMock stopMocking];
+}
+
+/** Tests that app delegate proxy is enabled when there is neither the Firebase nor the non-Firebase
+ *  Info.plist key present.
+ */
+- (void)testAppProxyPlistFlag_NoSceneDelegateProxyKey {
+  // No app delegate disable key. If there is no key, the default should be enabled.
+  NSDictionary *mainDictionary = @{@"randomKey" : @"randomValue"};
+  id mainBundleMock = OCMPartialMock([NSBundle mainBundle]);
+  [[[mainBundleMock expect] andReturn:mainDictionary] infoDictionary];
+
+  XCTAssertTrue([GULSceneDelegateSwizzler isSceneDelegateProxyEnabled]);
+  [mainBundleMock stopMocking];
```

Comment:
OC Mock might lead to memory leaks across multiple tests. Our recommendation is to always stop mocking and explicitly set the mock object to nil at the end of each unit test. We have seen tests crashed once in a while and after doing that, the flakiness went away. This is also recommended by readability reviewers.

Reply:
Good catch. fixed.